### PR TITLE
Connector service cancels a job during execution

### PR DESCRIPTION
**Pull request by wangch079**
## Closes https://github.com/elastic/enterprise-search-team/issues/3160

This PR made the changes to `SyncJobRunner` class:
1. to fail the job if 1) connector is deleted, or 2) sync job is deleted 3) sync job is not running (i.e. not in `in_progress` status)
2. to cancel the job if the job status is `canceling`.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

**Comment by artem-shelkovnikov** [comment]
Should this PR be based on top of one of other PRs @wangch079 ?

**Comment by wangch079** [comment]
>Should this PR be based on top of one of other PRs @wangch079 ?

Yes, I'm working on it now.

File: connectors/sync_job_runner.py [hunk 1]
```diff
@@ -143,20 +167,28 @@ async def _sync_done(self, sync_status, sync_error=None):
...
-        if sync_status == JobStatus.ERROR:
-            await self.sync_job.fail(sync_error, ingestion_stats=ingestion_stats)
-        elif sync_status == JobStatus.SUSPENDED:
-            await self.sync_job.suspend(ingestion_stats=ingestion_stats)
-        elif sync_status == JobStatus.CANCELED:
-            await self.sync_job.cancel(ingestion_stats=ingestion_stats)
-        else:
-            await self.sync_job.done(ingestion_stats=ingestion_stats)
+        if await self.reload_sync_job():
+            if await self.reload_connector():
```
Comment (artem-shelkovnikov) [inline connectors/sync_job_runner.py:173]: What will be "else" in this case? Do we want to log something in "else"?

File: connectors/sync_job_runner.py [hunk 2]
```diff
@@ -143,20 +167,28 @@ async def _sync_done(self, sync_status, sync_error=None):
...
+                await self.sync_job.suspend(ingestion_stats=ingestion_stats)
+            elif sync_status == JobStatus.CANCELED:
+                await self.sync_job.cancel(ingestion_stats=ingestion_stats)
+            else:
+                await self.sync_job.done(ingestion_stats=ingestion_stats)
+
+            await self.reload_sync_job()
+
+        if await self.reload_connector():
+            await self.connector.sync_done(self.sync_job)
```
Comment (artem-shelkovnikov) [inline connectors/sync_job_runner.py:190]: Same, what about `else`?

**Review by artem-shelkovnikov — APPROVED** [review]
Looks good, just left a couple questions

File: connectors/sync_job_runner.py (hunk 1 above)
Comment (wangch079) [reply connectors/sync_job_runner.py:173]: Make sense. Will add a log.

File: connectors/sync_job_runner.py (hunk 2 above)
Comment (wangch079) [reply connectors/sync_job_runner.py:190]: `else` means the connector is somehow deleted, so `self.connector.sync_done` will not be called. Will add a log for `else`.

File: connectors/sync_job_runner.py (hunk 1 above)
Comment (artem-shelkovnikov) [reply connectors/sync_job_runner.py:187]: Are you planning to add a log line here?

File: connectors/sync_job_runner.py (hunk 2 above)
Comment (artem-shelkovnikov) [reply connectors/sync_job_runner.py:204]: Same remark, I think it's worth to add a log line here

**Review by artem-shelkovnikov — APPROVED** [review]

File: connectors/sync_job_runner.py (hunk 1 above)
Comment (wangch079) [reply connectors/sync_job_runner.py:187]: I added the log to `reload_connector` method:

https://github.com/elastic/connectors-python/blob/06325cc0e563317fa296dcc05d9f2c7efa887353/connectors/sync_job_runner.py#L264-L272

But I just realized it won't always log when it returns `False`. I will add a log soon.

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**